Documentation/Using failure models.md

```diff
@@ -0,0 +1,83 @@
+# Using failure model example
+
+In some cases your request might return a server error, but you may still want to parse the data into a failure model.
```

Comment:
maybe we should define what you mean by a server error, i.e. saying a 4xx or 5xx HTTP status code